utils/cloudevent.go

```diff
@@ -50,8 +50,9 @@ func writeError(ceOut *CloudEventOut, err error) {
 	ceOut.EventTime = time.Now()
 }
 
-func DoCloudEventOnce(handler Handler, ctx context.Context, in io.Reader, out io.Writer, buf *bytes.Buffer, hdr http.Header) error {
+func DoCloudEventOnce(handler Handler, ctx context.Context, in io.Reader, out io.Writer, buf, inBody *bytes.Buffer, hdr http.Header) error {
```

Comment:
you can re-use the buffer to decode the input, reset it and hand it off to the handler. no need for 2. please remove

Reply:
done